checker/check.go

```diff
@@ -139,6 +147,15 @@ func checkProfile(p *proto.Profile, req *requestCache) (action Action) {
 }
 
 func checkRules(rules []*proto.Rule, req *requestCache, policyNamespace string) (action Action) {
+	defer func() {
+		// match checks data from dataplane which may be invalid.
+		// This is represented by the match function throwing a panic.
+		// Recover from the panic and perculate it up to the caller as INVALID_DATA
+		if r := recover(); r != nil {
+			log.Error("Match paniced; return INVALID_DATA")
```

Comment:
This will catch any `panic()`, not just the one you added for invalid data.  We want to only catch certain errors which are known to be recoverable.

I like to call this Pokemon exception handling (gotta catch 'em all), and it is a code smell :-)